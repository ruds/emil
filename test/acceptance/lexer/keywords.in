and anda as-cas
cases case
datatype else end exception
fn fun handle if implicit in infix infixr infixl
let local nonfix of open prefix raise rec then type
val while withtype : ::++:: _ __ _::_
| || |-| = => <= <=> -> --> # ## #=
