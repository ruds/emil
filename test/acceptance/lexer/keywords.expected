0001 KW_AND and
0001 ID_WORD 616E6461 anda
0001 KW_AS as
0001 ID_OP 2D -
0001 ID_WORD 636173 cas
0002 ID_WORD 6361736573 cases
0002 KW_CASE case
0003 KW_DATATYPE datatype
0003 KW_ELSE else
0003 KW_END end
0003 KW_EXCEPTION exception
0004 KW_FN fn
0004 KW_FUN fun
0004 KW_HANDLE handle
0004 KW_IF if
0004 KW_IMPLICIT implicit
0004 KW_IN in
0004 KW_INFIX infix
0004 KW_INFIXR infixr
0004 ID_WORD 696E6669786C infixl
0005 KW_LET let
0005 KW_LOCAL local
0005 KW_NONFIX nonfix
0005 KW_OF of
0005 KW_OPEN open
0005 KW_PREFIX prefix
0005 KW_RAISE raise
0005 KW_REC rec
0005 KW_THEN then
0005 KW_TYPE type
0006 KW_VAL val
0006 KW_WHILE while
0006 KW_WITHTYPE withtype
0006 COLON :
0006 ID_OP 3A3A2B2B3A3A ::++::
0006 KW_UNDERSCORE _
0006 ID_WORD 5F5F __
0006 KW_UNDERSCORE _
0006 ID_OP 3A3A ::
0006 KW_UNDERSCORE _
0007 PIPE |
0007 ID_OP 7C7C ||
0007 ID_OP 7C2D7C |-|
0007 EQUALS =
0007 TO_EXPR =>
0007 ID_OP 3C3D <=
0007 ID_OP 3C3D3E <=>
0007 TO_TYPE ->
0007 ID_OP 2D2D3E -->
0007 HASH #
0007 ID_OP 2323 ##
0007 ID_OP 233D #=
0008 END_OF_FILE
